src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs

```diff
@@ -51,7 +51,12 @@ public static SafeMsQuicConfigurationHandle Create(QuicClientConnectionOptions o
                 }
             }
 
-            return Create(options, QUIC_CREDENTIAL_FLAGS.CLIENT | QUIC_CREDENTIAL_FLAGS.USE_SUPPLIED_CREDENTIALS, certificate: certificate, certificateContext: null, options.ClientAuthenticationOptions?.ApplicationProtocols, options.ClientAuthenticationOptions?.CipherSuitesPolicy);
+            QUIC_CREDENTIAL_FLAGS flags = QUIC_CREDENTIAL_FLAGS.CLIENT;
+            if (OperatingSystem.IsWindows())
```

Comment:
as long as .NET doesn't use OpenSSL on Windows, this is fine.

Reply:
Do we have any way to query what backend the library uses? Otherwise having the flag produce errors with OpenSSL makes it a bit inconvenient.

Reply:
I don't think we have a way to query which crypto library is in use. @ThadHouse @nibanks, do you know of something?

Reply:
No, we don't currently expose a way to do this. We could via a global GetParam option easily enough. If you need this @rzikm please open a MsQuic issue for it.

Reply:
I still feel that making sure the flag does not break OpenSSL would be better option e.g. less platform code to maintain elsewhere.

Reply:
.NET uses the SChannel version on Windows, so unless we care that the user may somehow provide a different dll (possibly built with OpenSSL), then we don't need it.

Reply:
We have had asks to support OpenSSL MsQuic on Windows with .NET, so would prefer not to make assumptions of using Schannel on Windows.

Reply:
agreed. While we may not support it we may take community contributions. Requirement for S2022 or W11 seems pretty steep for many users.

Reply:
We also tie to `OperatingSystem.IsWindows()` (non) usage of portable certs. I assume that is also rather thing of OpenSSL and following would work with it:
https://github.com/dotnet/runtime/blob/cb1fd548e6bec61da18b75ae97cdad8d94b90c5f/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs#L178-L183

Unless, I'm mistaken here, I wouldn't put too much effort into distinguishing Windows from SChannel atm.

Reply:
I filed https://github.com/microsoft/msquic/issues/2761, it is not urgent, but it would be nice to have it in place when somebody eventually asks for support of OpenSSL-MsQuic on Windows.